cranelift/codegen/src/machinst/lower.rs

```diff
@@ -1325,7 +1340,14 @@ impl<'func, I: VCodeInst> Lower<'func, I> {
 impl<'func, I: VCodeInst> Lower<'func, I> {
     /// Get a new temp.
     pub fn alloc_tmp(&mut self, ty: Type) -> ValueRegs<Writable<Reg>> {
-        writable_value_regs(self.vregs.alloc(ty).unwrap())
+        let allocated = match self.vregs.alloc(ty) {
+            Ok(x) => x,
+            Err(e) => {
+                self.vreg_alloc_error = Some(e);
+                self.vregs.invalid(ty)
+            }
+        };
+        writable_value_regs(allocated)
```

Comment:
Could this method return a proper `Result` and then we have another `alloc_tmp_deferred_error` method that does the stuff-the-error-on-the-side bits? Then we could propagate the result normally from most callers and just the ISLE callers could use the deferral mechanism.

Reply:
I thought about this a bit but it seemed simpler to me to keep the one entry point -- as a method on `Lower` it's the one way to allocate temps during the main lowering loop, and we check during that loop for errors.